python/tank_vendor/shotgun_base/utils.py

```diff
@@ -18,6 +18,34 @@
 from .log import get_shotgun_base_logger
 log = get_shotgun_base_logger()
 
+
+def with_cleared_umask(func):
```

Comment:
i think that's overkill. This decorator is trying to address the common case which we have in lots of places in the code.